packages/compass-collection/src/stores/index.ts

```diff
@@ -248,13 +250,16 @@ store.onActivated = (appRegistry: AppRegistry) => {
 
   /**
    * Set the data service in the store when connected.
-   *
-   * @param {Error} error - The error.
-   * @param {DataService} dataService - The data service.
    */
-  appRegistry.on('data-service-connected', (error, dataService) => {
-    store.dispatch(dataServiceConnected(error, dataService));
-  });
+  appRegistry.on(
+    'data-service-connected',
+    (error, dataService: DataService) => {
+      store.dispatch(dataServiceConnected(error, dataService));
+      void dataService.instance().then(({ dataLake: { isDataLake } }) => {
+        store.dispatch(dataLakeChanged(isDataLake));
+      });
+    }
+  );
```

Comment:
Hmmm, that's weird that you have to do this here at all. But also I think there is an issue that seems to be in Compass for awhile where `isDataLake` is not passed to compass-collection plugin correctly. I do think thought that we should resolve this without making yet another call to `dataService.instance` outside the instance model that already has this information.

As this is something that can be addressed independently and the feature is behind a feature flag for now how do you feel about reverting the changes in this file (and keeping everything else you changed in compass-collection) and I can take a deeper look at this in parallel while you are wrapping up work on this PR?

Reply:
thanks for helping me with its resolution.
```
instance.dataLake.on(
  'change:isDataLake',
  (_model: unknown, value: boolean) => {
    store.dispatch(dataLakeChanged(value));
   }
);
```